app/src/main/java/campus/tech/kakao/map/MainActivity.kt

```diff
@@ -1,11 +1,143 @@
 package campus.tech.kakao.map
 
+import android.content.SharedPreferences
 import android.os.Bundle
+import android.view.View
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.DefaultItemAnimator
+import androidx.recyclerview.widget.LinearLayoutManager
+import campus.tech.kakao.map.model.RecentSearchWord
+import campus.tech.kakao.map.databinding.ActivityMainBinding
+import campus.tech.kakao.map.viewModel.MapRepository
+import campus.tech.kakao.map.viewModel.PlacesViewModel
+import campus.tech.kakao.map.viewModel.PlacesViewModelFactory
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var repository: MapRepository
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var viewModelFactory: PlacesViewModelFactory
+    private lateinit var viewModel: PlacesViewModel
+    private lateinit var placesAdapter: PlacesAdapter
+
+    private lateinit var prefs: SharedPreferences
+    private lateinit var prefEditor: SharedPreferences.Editor
+    private var stringPrefs: String? = null
+    private var searchHistoryList = ArrayList<RecentSearchWord>()
+    private lateinit var searchHistoryAdapter: SearchHistoryAdapter
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setPrefs()
+        setUpSearchHistoryAdapter()
+        setUpPlacesAdapter()
+
+        repository = MapRepository(this)
+        viewModelFactory = PlacesViewModelFactory(repository)
+        viewModel = ViewModelProvider(this, viewModelFactory).get(PlacesViewModel::class.java)
+
+        setUpViewModelObservers()
+
+        binding.searchInput.addTextChangedListener { text ->
+            viewModel.filterPlace(text.toString())
+        }
+
+        binding.deleteInput.setOnClickListener {
+            binding.searchInput.text.clear()
+        }
+
+        updateSearchHistoryVisibility()
+    }
+
+    private fun setUpSearchHistoryAdapter() {
+        searchHistoryAdapter = SearchHistoryAdapter(
+            searchHistoryList,
+            { position: Int ->
+                delSearch(position)
+                updateSearchHistoryVisibility()
+            },
+            { position: Int ->
+                val itemName = searchHistoryAdapter.getItemName(position)
+                binding.searchInput.setText(itemName)
+            })
```

Comment:
요럴 땐 argument name을 함께 붙여주시는 걸 추천드리고 싶어요! 
파라미터를 바꿔 작성할 실수도 줄이고, 각각의 람다가 무엇을 위한 역할인지도 확실하게 할 수 있는 장점이 있습니다.

```suggestion
            onDeleteClick = { position: Int ->
                delSearch(position)
                updateSearchHistoryVisibility()
            },
            onTextClick = { position: Int ->
                val itemName = searchHistoryAdapter.getItemName(position)
                binding.searchInput.setText(itemName)
            })
```